src/components/Booking/Booking.js

```diff
@@ -52,84 +60,125 @@ function Booking() {
         }
     }
 
-
-    const first_arr = parkSpots.slice(0, 10);
-    const second_arr = parkSpots.slice(10, 20);
-    const third_arr = parkSpots.slice(20, 30);
+    const closeModal = () => {
+        setShowModal(false)
+        navigate(BOOKING_PATH);
+    }
 
     const onSelectSpot = (parkSpot) => {
-        let newParkSpots = parkSpots.map(p => {
+        let updatedParkSpots = parkSpots.map(p => {
             let newP = {...p, isSelected: false};
             if (newP.isOccupied) {
                 return newP;
             }
             return newP.id === parkSpot.id ? {...newP, isSelected: true} : newP;
+
         });
-        setParkSpots(newParkSpots)
+        setParkSpots(updatedParkSpots)
+        setSpaceName(parkSpot.space_name)
     }
 
-    const closeModal = () => {
-        setShowModal(false)
-        navigate(BOOKING_PATH);
+    const sendBookingRequest = async () => {
+        try {
+            const response = await axios.post(BOOKING_ENDPOINT, {
+                space_name: spaceName,
+                check_in: checkOutTime,
+                check_out: checkOutTime,
+                price: price
+            }, {
+                headers: {token}
+            })
+            setBookingSuccessful(true)
+            console.log(response)
+        } catch (error) {
+            setBookingSuccessful(false)
+            console.log(error.response)
+            const errorResponse = error.response
+            if (errorResponse.status === 401) {
+                setErrorMessage("Access denied. You're unauthorised. Please sign out and sign in.")
+            } else if (errorResponse.status === 400) {
+                setErrorMessage("Choose a park space and select a date or time range. Please try again")
+            } else {
+                setErrorMessage("There was an issue making booking, please try again")
+            }
+        }
+    }
+
+    const handleSubmitBooking = async () => {
+        if (canSubmitBooking) {
+            setShowModal(true);
```

Comment:
you shouldn't do this, you should only show the modal when there's an error from the backend or when the booking was successful. So, I'd suggest you do this in the sendBookingRequest method